hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/filedrops/ResponseFileDropGenerater.java

```diff
@@ -0,0 +1,85 @@
+package ca.bc.gov.hlth.hnsecure.filedrops;
+
+
+import static ca.bc.gov.hlth.hnsecure.properties.ApplicationProperty.FILE_DROPS_LOCATION;
+import static java.nio.file.StandardOpenOption.CREATE;
+
+import java.io.BufferedOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.PrintWriter;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import org.apache.camel.Exchange;
+import org.apache.camel.Handler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ca.bc.gov.hlth.hnsecure.parsing.Util;
+import ca.bc.gov.hlth.hnsecure.properties.ApplicationProperties;
+
+/**
+ * This implementation logs each HL7v2 request/response message in a file in its raw/text form.
+ * file name format:{messageid}-{messagetype}-{facilityid}-{messagedate}-{request/response}.txt
+ * @author anumeha.srivastava
+ *
+ */
+public class ResponseFileDropGenerater {
```

Comment:
There is quite a bit of duplicated code in the request and response file drop classes. I think you could probably have an abstract class and then have the implementations just set request or response

Reply:
Done.